CountingSort/app_resources/prefix_sum_shader.comp.hlsl

```diff
@@ -0,0 +1,16 @@
+#include "nbl/builtin/hlsl/sort/counting.hlsl"
+
+[[vk::push_constant]] nbl::hlsl::sort::CountingPushData pushData;
+
+uint32_t3 nbl::hlsl::glsl::gl_WorkGroupSize()
+{
+    return uint32_t3(WorkgroupSize, 1, 1);
+}
+
+[numthreads(WorkgroupSize,1,1)]
+void main(uint32_t3 ID : SV_GroupThreadID, uint32_t3 GroupID : SV_GroupID)
+{
+    nbl::hlsl::sort::counting < bda::__ptr < uint32_t >, bda::__ptr < uint32_t >, bda::__ptr < uint32_t > > counter;
```

Comment:
design should be slightly different, don't use BDA pointers directly, lets have Accessors

Accessor pattern is important because it lets you use any underlying resource via static polymorphism"
- bda pointers
- ssbos/ubos
- sampler buffers
- textures
- shared memory

Reply:
You mean rewrite the histogram() and scatter() functions to be completely generic?

Reply:
not functions.

Accessor is a concept, e.g.
```cpp
struct MyAccessor
{
    uint32_t get(uint64_t index);
    void set(uint64_t index, uint32_t value);
};
```

If you use a `MyAccessor` that has those methods, the utility should work.